src/get_trigger.ts

```diff
@@ -1,23 +1,25 @@
 import { parse, path } from "./deps.ts";
+import { getProtocolInterface } from "./protocol/mod.ts";
+import type { Protocol } from "./protocol/types.ts";
 
-const getTrigger = async () => {
-  const source = parse(Deno.args).source as string;
+const getTrigger = async (args: string[], hookCLI: Protocol) => {
+  const source = parse(args).source as string;
```

Comment:
I moved parsing of CLI arguments out of this function and into the `import.meta.main` conditional at the bottom, to follow the same pattern as the other hook implementations. This makes testing this function easier in unit tests, as you can simply pass CLI flags to the function to test different behaviours, rather than have to invoke the test file with CLI arguments.